cumulus/tasks/pdr-status-check/index.js

```diff
@@ -1,123 +1,177 @@
-/* eslint-disable no-param-reassign */
 'use strict';
 
-const get = require('lodash.get');
-const { StepFunction } = require('@cumulus/ingest/aws');
+const cumulusMessageAdapter = require('@cumulus/cumulus-message-adapter-js');
+const aws = require('@cumulus/common/aws');
 const { IncompleteError } = require('@cumulus/common/errors');
 let log = require('@cumulus/ingest/log');
 
 log = log.child({ file: 'pdr-status-check/index.js' });
 
+// The default number of times to re-check for completion
+const defaultRetryLimit = 30;
+
 /**
-* Callback function provided by aws lambda. See https://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-handler.html#nodejs-prog-model-handler-callback
-* @callback lambdaCallback
-* @param {object} error
-* @param {object} output - output object matching schemas/output.json
-*/
+ * Determine the number of times this check has been performed
+ *
+ * If not set, defaults to 0.
+ *
+ * @param {Object} event - a simple Cumulus event
+ * @returns {integer} - the number of times this check has been run
+ */
+const getCounterFromEvent = (event) => event.input.counter || 0;
```

Comment:
This will cause error if event doesn't have input.

Reply:
If there isn't input then there's a bug in the cumulus-message-adapter.